### PR TITLE
Upgrade SBT to 1.5.5

### PR DESCRIPTION
**Pull request by scottsand-db**
This PR takes over the SBT upgrade work from #642 . Thanks Jacek Laskowski for the initial investigation and great work!

All pass:
- `build/sbt compile`
- `build/sbt package`
- `build/sbt test`
- `build/sbt unidoc`

And (in `examples/scala`
- `build/sbt package`
- `build/sbt compile`

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -14,30 +14,16 @@
...
-addSbtPlugin("com.github.sbt" % "sbt-release" % "1.0.15")
-
-addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "2.3")
-
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
-
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.9")
-
-addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.3")
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "1.0.0")
```
Comment (zsxwing) [inline project/plugins.sbt:21]: I think we can remove this

File: project/plugins.sbt [hunk 2]
```diff
@@ -14,30 +14,16 @@
...
-
-addSbtPlugin("com.github.sbt" % "sbt-release" % "1.0.15")
-
-addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "2.3")
-
-addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
-
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.9")
-
-addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.3")
```
Comment (zsxwing) [inline project/plugins.sbt:32]: I think we still need other plugins except `sbt-coursier` which should is already in SBT 1.5. Those plugins are for release process.

File: .gitignore [hunk 3]
```diff
@@ -8,6 +8,7 @@
 *.swp
 *~
 .DS_Store
+.bsp/
```
Comment (zsxwing) [inline .gitignore:11]: what's in this directory?

File: build/sbt-launch-lib.bash [hunk 4]
```diff
@@ -38,12 +38,11 @@ dlog () {
 
 acquire_sbt_jar () {
```
Comment (zsxwing) [inline build/sbt-launch-lib.bash:39]: Where did you get this file, or you just manually modified it?

**Review by zsxwing — CHANGES_REQUESTED** [review]
Cool! Left some minor comments. Could you also update SBT in https://github.com/delta-io/delta/tree/master/examples/scala ?

File: build/sbt-launch-lib.bash (hunk 4 above)
Comment (scottsand-db) [reply build/sbt-launch-lib.bash:39]: https://github.com/delta-io/delta/pull/642/files#diff-3d92b4f5f048a351dd0984abcc67b6bc54fffb54aaf8f505d5f95ce800d36e11

File: project/plugins.sbt (hunk 2 above)
Comment (scottsand-db) [reply project/plugins.sbt:32]: i have added back `addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "2.3")`
that was the only plugin missing (i.e. we have release, sonatype, pgp, assembly now, as required)

File: .circleci/config.yml [hunk 5]
```diff
@@ -2,7 +2,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: circleci/openjdk:8u181-jdk # java 8
```
[the remark this answers is not in the document]
Comment (scottsand-db) [reply .circleci/config.yml:5]: fails without it
https://app.circleci.com/pipelines/github/delta-io/delta/1096/workflows/f5ea2fed-4185-4075-904e-0c2fd883acd3/jobs/1111

File: .gitignore (hunk 3 above)
Comment (scottsand-db) [reply .gitignore:11]: removed

File: .circleci/config.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (scottsand-db) [reply .circleci/config.yml:5]: and with the change, it does not fail: https://app.circleci.com/pipelines/github/delta-io/delta/1097/workflows/6f3e7dec-a303-4238-a4f6-25ea96c36141/jobs/1112

**Review by zsxwing — APPROVED** [review]

**Comment by jaceklaskowski** [comment]
LGTM

**Review by jaceklaskowski — APPROVED** [review]

**Pull request closed without merging**